open_schools_platform/history_management/views.py

```diff
@@ -21,111 +28,202 @@
     StudentProfileHistorySerializer
 from open_schools_platform.history_management.serializers.parent_serializer import ParentProfileHistorySerializer
 from open_schools_platform.history_management.serializers.family_serializer import FamilyHistorySerializer
+from rest_framework.generics import ListAPIView
 
 
-class UserHistoryApi(ApiAuthMixin, APIView):
+class UserHistoryApi(ApiAuthMixin, ListAPIView):
+    queryset = User.objects.all()
+    pagination_class = ApiHistoryListPagination
+    serializer_class = UserHistorySerializer
+
     @swagger_auto_schema(
         operation_description="Get user history",
         tags=[SwaggerTags.HISTORY_MANAGEMENT],
         responses={200: convert_dict_to_serializer({'results': UserHistorySerializer()}),
                    404: "No such user"},
     )
     def get(self, request, pk):
-        user = get_user(filters={"id": pk}, user=request.user, empty_exception=True)
-        return Response({"results": UserHistorySerializer(user).data}, status=200)
+        response = get_paginated_response(
+            pagination_class=ApiHistoryListPagination,
+            serializer_class=UserHistorySerializer,
+            queryset=get_user(filters={"id": pk}, user=request.user, empty_exception=True).history.all(),
```

Comment:
we should write empty_exception=False because there are lists of objects

Reply:
There is a one user and if it doesn't exists then its 404